src/digital.rs

```diff
@@ -116,14 +135,30 @@ pub trait ToggleableOutputPin {
 ///    }
 /// }
 ///
+/// #[cfg(feature = "use-fallible-digital-traits")]
+/// impl StatefulOutputPin for MyPin {
+///    fn is_set_low(&self) -> Result<bool, Self::Error> {
+///        Ok(!self.state)
+///    }
+///    fn is_set_high(&self) -> Result<bool, Self::Error> {
+///        Ok(self.state)
+///    }
+/// }
+///
 /// /// Opt-in to the software implementation.
 /// impl toggleable::Default for MyPin {}
 ///
 /// let mut pin = MyPin { state: false };
 /// pin.toggle().unwrap();
+/// #[cfg(not(feature = "use-fallible-digital-traits"))]
 /// assert!(pin.is_set_high());
+/// #[cfg(feature = "use-fallible-digital-traits")]
+/// assert!(pin.is_set_high().unwrap());
 /// pin.toggle().unwrap();
+/// #[cfg(not(feature = "use-fallible-digital-traits"))]
 /// assert!(pin.is_set_low());
+/// #[cfg(feature = "use-fallible-digital-traits")]
+/// assert!(pin.is_set_low().unwrap());
```

Comment:
Hmm, I'm wondering, if we should simplify the example. I mean, we want people to use the new fallible versions, so why even show the infallible versions here? Thoughts?

Reply:
I agree that the docs should prioritise the fallible version, though maybe we could keep the infallible documentation until the opt-out release?

Reply:
I moved the fallible versions up. However, the example still looks quite cluttered with `#cfg...` to me. Any ideas?

Reply:
It is a bit cluttered, maybe grouping each example would help? 
I'm ok with merging this either way.

Reply:
I found no way to group the statements so that I can have fewer `#cfg`s. Do you have an idea?